libs/vertexai/langchain_google_vertexai/vectorstores/feature_store/executors.py

```diff
@@ -0,0 +1,956 @@
+from __future__ import annotations
+
+import time
+from datetime import datetime, timedelta
+from subprocess import TimeoutExpired
+from threading import Lock, Thread
+from typing import Any, Dict, List, Literal, MutableSequence, Optional, Union
+
+import numpy as np
+import proto  # type: ignore[import-untyped]
+import vertexai  # type: ignore[import-untyped]
+from google.api_core.exceptions import (
+    ClientError,
+    MethodNotImplemented,
+    ServiceUnavailable,
+)
+from google.cloud import bigquery
+from google.cloud.aiplatform import base, telemetry
+from google.cloud.aiplatform_v1beta1 import (
+    FeatureOnlineStoreAdminServiceClient,
+    FeatureOnlineStoreServiceClient,
+)
+from google.cloud.aiplatform_v1beta1.types import (
+    NearestNeighborQuery,
+    feature_online_store_service,
+)
+from google.cloud.aiplatform_v1beta1.types import (
+    feature_online_store as feature_online_store_pb2,
+)
+from langchain_core.documents import Document
+from pydantic import BaseModel, ConfigDict
+from vertexai.resources.preview import (  # type: ignore[import-untyped]
+    AlgorithmConfig,
+    DistanceMeasureType,
+    FeatureOnlineStore,
+    FeatureView,
+    FeatureViewBigQuerySource,
+)
+from vertexai.resources.preview.feature_store import (  # type: ignore[import-untyped]
+    utils,
+)
+
+from langchain_google_vertexai._utils import get_client_info, get_user_agent
+from langchain_google_vertexai.vectorstores.feature_store.utils import (
+    EnvConfig,
+    cast_proto_type,
+    doc_match_filter,
+)
+
+_vector_table_lock = Lock()  # process-wide BigQueryVectorSearch table lock
+
+logger = base.Logger(__name__)
+# Constants for index creation
+MIN_INDEX_ROWS = 5
+INDEX_CHECK_INTERVAL = timedelta(seconds=60)
+USER_AGENT_PREFIX = "FeatureStore"
+
+
+class BaseExecutor(BaseModel):
+    _env_config: EnvConfig = EnvConfig()
+    _user_agent: str = get_user_agent(None)[1]
+
+    def sync(self):
+        raise NotImplementedError()
+
+    def similarity_search_by_vectors_with_scores_and_embeddings(
+        self,
+        embeddings: List[List[float]],
+        filter: Optional[Dict[str, Any]] = None,
+        k: int = 5,
+        batch_size: Union[int, None] = None,
+    ) -> list[list[list[Any]]]:
+        raise NotImplementedError()
+
+    def set_env_config(self, env_config: EnvConfig):
+        self._env_config = env_config
+
+
+class BruteForceExecutor(BaseExecutor):
```

Comment:
why do we need our own in-memory implementation if LC already has a few? can we just write an adapter that will load data into it?